File: python/38.py
import unittest


class Solution:
    def countAndSay(self, n: int) -> str:
        ret = '1'
        for _ in range(n - 1):
            next = ''
            ch = ret[0]
            count = 1
            for i in range(1, len(ret)):
                if ret[i] == ch:
                    count += 1
                    continue

                next = f'{next}{count}{ch}'
                ch = ret[i]
                count = 1
            ret = f'{next}{count}{ch}'
        return ret


class TestSolution(unittest.TestCase):
    def test(self):
        solution = Solution()
        self.assertEqual(solution.countAndSay(4), '1211')

File: typescript/38.ts
function countAndSay(n: number): string {
    let ret = '1';
    for (let i = 1; i < n; i++) {
        let next = '';
        let ch = ret[0];
        let count = 1;
        for (let j = 1; j < ret.length; j++) {
            if (ret[j] === ch) {
                count++;
                continue;
            }

            next = `${next}${count}${ch}`;
            ch = ret[j];
            count = 1;
        }
        ret = `${next}${count}${ch}`;
    }
    return ret;
};

test('38', () => {
    expect(countAndSay(4)).toBe('1211');
});